reroll-initiative/old/reroll-initiative.js

```diff
@@ -0,0 +1,260 @@
+/**
+ * @name Reroll-Initiative
+ * @version 0.4
+ * @author Evan Clarke <errational>
+ * @description Rerolls initiative on combat round change
+ */
+
+/**
+ * @class RerollInitiative
+ * @description Hooks on combat update and rerolls initiative for all combatants depending on current value of setting
+ * @todo Add configurability for whom to reroll, make enable setting define whether the hook is registered or not??
+ */
+class RerollInitiative {
+
+    constructor() {
+        /**
+         * Initialize object to hold the config for this instance
+         * @type {Object}
+         */
+        this.config = {};
+
+        this._registerSettings();
+
+        this._loadSettings();
+
+        this._postUpdateCombatHook();
+    }
+
+    /**
+     * Define the default config for the module. Restricted to get only
+     */
+    static get DEFAULT_CONFIG() {
+        return {
+            reroll: true,
+            actorTypes: "all" //future feature
+        };
+    }
+
+    /**
+     * Define the settings metadata for the module. Restricted to get only
+     */
+    static get SETTINGS() {
+        return {
+            module: "reroll-initiative",
+            key: "rriSettings",
+            name: "Reroll-Initiative Settings",
+            hint: "Settings for Reroll-Initiative module",
+            default: RerollInitiative.DEFAULT_CONFIG,
+            type: Object,
+            scope: "world"
+        };
+    }
+
+    /**
+     * Register module settings with game settings
+     * @todo: Add any steps to occur when settings change
+     */
+    _registerSettings () {
+        game.settings.register(RerollInitiative.SETTINGS.module, RerollInitiative.SETTINGS.key, {
+            name: RerollInitiative.SETTINGS.name,
+            hint: RerollInitiative.SETTINGS.hint,
+            default: RerollInitiative.SETTINGS.default,
+            type: RerollInitiative.SETTINGS.type,
+            scope: RerollInitiative.SETTINGS.scope,
+            onChange: settings => {
+                
+                console.log("Module settings changed, new option values: ",settings)
+            }
+        });
+    }
+
+    /**
+     * Resets settings back to default
+     * @todo: maybe expand this to deregister and register settings
+     */
+    _applydefaultConfig() {
+        this.config = RerollIinitiative.DEFAULT_CONFIG;
+        console.log("Resetting reroll-initiative settings to defaults:",RerollInitiative.DEFAULT_CONFIG);
+        this._saveSettings();
+    }
+
+    /**
+     * Saves current class instance options back to game settings
+     */
+    async _saveSettings () {
+        await game.settings.set(RerollInitiative.SETTINGS.module,RerollInitiative.SETTINGS.key,this.config);
```

Comment:
Minor: I always like from a readability aspect to new line arguments when there are more than 2. It's easier to scan the code and understand. Completely a preference thing.